shaders/clouds.omwfx

```diff
@@ -472,33 +449,33 @@ fragment main {
     vec3 sun_dir = normalize(omw.sunPos.xyz);
 
     vec3 sun_color = mix(
-        vec3(2, 0.5, 0.2) * 3,
+        vec3(2.0, 0.5, 0.2) * 3.0,
         vec3(0.6, 0.5, 0.0),
         pow(max(sun_dir.z + 0.2, 0.001), 0.1)
     );
 
-	vec3 weather_tone[10]=vec3[](
-		vec3(0.85),  //Clear
-		vec3(0.7), //Cloudy
-		vec3(0.65),  //Foggy
-		vec3(0.6),  //Overcast
-		vec3(0.55),  //Rain
-		vec3(0.5, 0.5, 0.6),  //Thunder
-		vec3(1.0, 0.5, 0.25),  //Ash
-		vec3(1.0, 0.5, 0.5),  //Blight
-		vec3(1.0),   //Snow
-		vec3(1.0)   //Blizzard
-	);
-
-    vec3 sky_tone = mix(weather_tone[omw.weatherID], weather_tone[omw.nextWeatherID], omw.weatherTransition);
+    vec3 weather_tone(int i) {
+        if( i == 0) return vec3(0.85);  //Clear
+        if( i == 1) return vec3(0.7); //Cloudy
+        if( i == 2) return vec3(0.65);  //Foggy
+        if( i == 3) return vec3(0.6);  //Overcast
+        if( i == 4) return vec3(0.55);  //Rain
+        if( i == 5) return vec3(0.5, 0.5, 0.6);  //Thunder
+        if( i == 6) return vec3(1.0, 0.5, 0.25);  //Ash
+        if( i == 7) return vec3(1.0, 0.5, 0.5);  //Blight
+        if( i == 8) return vec3(1.0);   //Snow
+        return vec3(1.0);   //Blizzard
+    }
```

Comment:
I am slightly nervous about how well this optimises in practice. That said, the rest of the shader is substantially more expensive, so I guess it's fine.